allensdk/brain_observatory/data_release_utils/metadata_utils/utils.py

```diff
@@ -63,15 +80,28 @@ def add_file_paths_to_metadata_table(
                f"{on_missing_file}")
         raise ValueError(msg)
 
-    file_suffix = 'nwb'
     new_data = []
     missing_files = []
     metadata_table = metadata_table.set_index(index_col)
 
     for row in metadata_table.itertuples():
-        session_id = getattr(row, session_id_col, row.Index)
-        file_path = file_dir / f'{session_id}' /  \
-            f'{file_prefix}_{row.Index}.{file_suffix}'
+        data_dir = getattr(row, data_dir_col, row.Index)
+
+        if file_stem is None:
+            file_stem_ = \
+                f'{file_prefix}_{row.Index}' if file_prefix is not None else \
+                f'{row.Index}'
+        else:
+            file_stem_ = file_stem
+
+        if data_dir is None:
+            # If session_id_col is not given, assume files stored flat
```

Comment:
"If session_id or similar column is not given, assume files stored flat."

Reply:
Reworded.